sdk/textanalytics/out/azure/ai/textanalytics/aio/_text_analytics_client_async.pyi

```diff
@@ -2,16 +2,18 @@
 #
 # NOTE: This dynamically typed stub was automatically generated by stubgen.
 
+from .._credential import TextAnalyticsApiKeyCredential
 from .._generated.models import TextAnalyticsErrorException
 from .._models import AnalyzeSentimentResult, DetectLanguageInput, DetectLanguageResult, DocumentError, ExtractKeyPhrasesResult, RecognizeEntitiesResult, RecognizeLinkedEntitiesResult, RecognizePiiEntitiesResult, TextDocumentInput
 from .._request_handlers import _validate_batch_input
 from .._response_handlers import entities_result, key_phrases_result, language_result, linked_entities_result, pii_entities_result, process_batch_error, sentiment_result
 from ._base_client_async import AsyncTextAnalyticsClientBase
+from azure.core.credentials_async import AsyncTokenCredential
 from typing import Any, Dict, List, Optional, Union
 
 class TextAnalyticsClient(AsyncTextAnalyticsClientBase):
-    def __init__(self, endpoint: str, credential: Any, **kwargs: Any) -> None: ...
-    async def detect_languages(self, inputs: Union[List[str], List[DetectLanguageInput], List[Dict[str, str]]], country_hint: Optional[str]=..., **kwargs: Any) -> List[Union[DetectLanguageResult, DocumentError]]: ...
+    def __init__(self, endpoint: str, credential: Union[TextAnalyticsApiKeyCredential, AsyncTokenCredential], **kwargs: Any) -> None: ...
+    async def detect_language(self, inputs: Union[List[str], List[DetectLanguageInput], List[Dict[str, str]]], country_hint: Optional[str]=..., **kwargs: Any) -> List[Union[DetectLanguageResult, DocumentError]]: ...
```

Comment:
Does the `country_hint` here apply to all the inputs in the list?
Are the inputs listed expected to be in the same language? Or are each evaluated independently?

Reply:
Good question. There are several ways a `country_hint` can be set for your batch, in order of what takes precedence:

1) Individually add a `country_hint` to each item in the batch by passing input as a list of dict, e.g. `[{"id": "1", "country_hint": "US", "text": "this is written in english"}, ...]`
2) Pass `country_hint` at the operation-level (each item will use this hint)
3) Set the `default_country_hint` at the client level (If user doesn't pass (1) or (2), then each item uses this hint)

The `language` concept for the other methods works the same way. The inputs do not need to be in the same language, but if a user does not specify a `country_hint` the service will use "US" as the default (most of their customer base). We recently added the option of passing a `country_hint="none"` if you don't want the service to use that default.